src/etc/inc/interfaces.inc

```diff
@@ -1156,6 +1142,60 @@ function interface_gre_configure(&$gre, $grekey = "") {
 	return $greif;
 }
 
+function interface_is_type($if = NULL, $type) {
+	global $config;
+	switch ($type) {
+		case "gre":
+			$list = 'gres';
+			$entry = 'gre';
+			$entif = 'greif';
+			break;
+		case "gif":
+			$list = 'gifs';
+			$entry = 'gif';
+			$entif = 'gifif';
+			break;
+		case "lagg":
+			$list = 'laggs';
+			$entry = 'lagg';
+			$entif = 'laggif';
+			break;
+		default:
+			break;
+	}
+
+	if (!isset($config[$list][$entry]) || !is_array($config[$list][$entry])) {
+		return (NULL);
+	}
+
+	foreach ($config[$list][$entry] as $ent) {
+		if ($ent[$entif] == $if) {
+			return ($ent);
+		}
+	}
+	return (NULL);
+}
+
+function is_greipsec($if) {
+	global $config;
+
+	if (ipsec_enabled() && is_array($config['gres']) && is_array($config['ipsec']['phase2'])) {
+		foreach ($config['gres']['gre'] as $gre) {
```

Comment:
`$config['gres']` may be an array (for the previous test) but that doesn't guarantee `$config['gres']['gre']` is. It either needs tested or setup with `init_config_arr()` before this foreach.

Reply:
is_array($config['gres']['gre']) added